File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashSet<Integer>occurance = new HashSet<>();
        for (int i : nums) { 
            if (occurance.contains(i)) {
                return true;
            } 
            occurance.add(i);
        }

        return false;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val valtoindex = mutableMapOf<Int, Int>()
        for (i in nums.indices) {
            val needed = target - nums[i]

            if (valtoindex.contains(needed)) {
                return intArrayOf(valtoindex[needed]!!, i)
            }

            valtoindex[nums[i]] = i
        }

        return intArrayOf()
    }
}